src/lib/redux-cozy-api.js

```diff
@@ -0,0 +1,484 @@
+/* global cozy */
+import { combineReducers } from 'redux'
+
+const FILES_DOCTYPE = 'io.cozy.files'
+const FETCH_LIMIT = 10
+
+// action types
+const REGISTER_SCHEMAS = 'REGISTER_SCHEMAS'
+const GET_OR_CREATE_INDEX = 'GET_OR_CREATE_INDEX'
+const FETCH_DOCUMENTS = 'FETCH_DOCUMENTS'
+const FETCH_DOCUMENT = 'FETCH_DOCUMENT'
+const FETCH_REFERENCED_FILES = 'FETCH_REFERENCED_FILES'
+const ADD_REFERENCED_FILES = 'ADD_REFERENCED_FILES'
+const REMOVE_REFERENCED_FILES = 'REMOVE_REFERENCED_FILES'
+const CREATE_ENTITY = 'CREATE_ENTITY'
+const UPDATE_ENTITY = 'UPDATE_ENTITY'
+const DELETE_ENTITY = 'DELETE_ENTITY'
+const RECEIVE_DATA = 'RECEIVE_DATA'
+const RECEIVE_DOCUMENT = 'RECEIVE_DOCUMENT'
+const RECEIVE_REFERENCED_FILES = 'RECEIVE_REFERENCED_FILES'
+const RECEIVE_CREATION_CONFIRM = 'RECEIVE_CREATION_CONFIRM'
+const RECEIVE_UPDATE_CONFIRM = 'RECEIVE_UPDATE_CONFIRM'
+const RECEIVE_DELETION_CONFIRM = 'RECEIVE_DELETION_CONFIRM'
+const RECEIVE_ERROR = 'RECEIVE_ERROR'
+const RECEIVE_CREATION_ERROR = 'RECEIVE_CREATION_ERROR'
+const RECEIVE_UPDATE_ERROR = 'RECEIVE_UPDATE_ERROR'
+const RECEIVE_DELETION_ERROR = 'RECEIVE_DELETION_ERROR'
+
+// reducers
+const schemas = (state = {}, action) => {
+  switch (action.type) {
+    case REGISTER_SCHEMAS:
+      return Object.assign({}, state, action.schemas)
+    case GET_OR_CREATE_INDEX:
+      return {
+        ...state,
+        [action.doctype]: Object.assign({}, state[action.doctype], { index: action.index })
+      }
+    default:
+      return state
+  }
+}
+
+const entities = (state = {}, action) => {
+  switch (action.type) {
+    case RECEIVE_DATA:
+      return {
+        ...state,
+        [action.doctype]: Object.assign({}, state[action.doctype], objectifyEntitiesArray(action.response.data))
```

Comment:
[destructuring](http://babeljs.io/learn-es2015/#ecmascript-2015-features-destructuring) and [rest, spread operators](http://babeljs.io/learn-es2015/#ecmascript-2015-features-default-rest-spread) are just very useful imho